commons/src/lib.rs

```diff
@@ -94,34 +106,59 @@ pub fn ensure_query_params(
 /// Make sure the client can accept the provided media type.
 pub fn validate_content_type(
     headers: &HeaderMap,
-    content_type: &'static str,
-) -> Result<(), GraphError> {
+    mut content_type: Vec<actix_web::http::HeaderValue>,
+    accept_default: actix_web::http::HeaderValue,
+) -> Result<String, GraphError> {
     let header_value = match headers.get(header::ACCEPT) {
-        None => return Ok(()),
+        None => {
+            let minimum_version = MIN_CINCINNATI_VERSION.to_string();
+            return Ok(minimum_version);
+        }
         Some(v) => v,
     };
 
-    let full_type = header::HeaderValue::from_static(content_type);
     let wildcard = header::HeaderValue::from_static("*");
     let double_wildcard = header::HeaderValue::from_static("*/*");
-    let top_type = content_type.split("/").next().unwrap_or("");
-    let top_type_wildcard = header::HeaderValue::from_str(&format!("{}/*", top_type));
-    assert!(
-        top_type_wildcard.is_ok(),
-        "could not form top-type wildcard from {}",
-        top_type
-    );
-
-    let acceptable_content_types: Vec<actix_web::http::HeaderValue> = vec![
-        full_type,
-        wildcard,
-        double_wildcard,
-        top_type_wildcard.unwrap(),
-    ];
+
+    let mut top_types: Vec<actix_web::http::HeaderValue> = content_type
+        .iter()
+        .map(|ct| {
+            let top_type = ct.to_str().unwrap_or("").split("/").next().unwrap_or("");
+            let top_type_wildcard = header::HeaderValue::from_str(&format!("{}/*", top_type));
+            assert!(
+                top_type_wildcard.is_ok(),
+                "could not form top-type wildcard from {}",
+                top_type
+            );
+            top_type_wildcard.unwrap()
+        })
+        .collect();
+
+    let mut acceptable_content_types: Vec<actix_web::http::HeaderValue> =
+        vec![wildcard, double_wildcard, accept_default.clone()];
+    acceptable_content_types.append(&mut content_type);
+    acceptable_content_types.append(&mut top_types);
 
     // FIXME: this is not a full-blown Accept parser
     if acceptable_content_types.iter().any(|c| c == header_value) {
-        Ok(())
+        return if header_value
+            .to_str()
+            .unwrap_or("")
+            .split("/")
+            .any(|i| i == "*")
+        {
+            Ok(header::HeaderValue::to_str(&accept_default)
+                .unwrap()
+                .parse()
+                .unwrap())
+        } else {
+            let minimum_version: String = MIN_CINCINNATI_VERSION.to_string();
+            let accept = header::HeaderValue::to_str(header_value);
+            match accept {
+                Ok(a) => Ok(a.parse().unwrap()),
+                Err(_e) => Ok(minimum_version),
```

Comment:
```suggestion
                Err(_e) => Ok(MIN_CINCINNATI_VERSION.to_string()),
```
same, extra `let` is not necessary